dqc_us_rules/dqc_us_0041.py

```diff
@@ -0,0 +1,294 @@
+
+    year = 2014
+
+    for ugt in ugtDocs:
+        ugt_namespace = ugt["namespace"]
+        if _is_in_namespace(val, ugt_namespace):
+            ugt_default_dimensions_json_file = os.path.join(
+                os.path.dirname(__file__),
+                'resources',
+                'DQC_US_0041',
+                "_".join([str(year), "ugt-default-dimensions.json"])
+            )
+
+            file = None
+
+            try:
+                file = openFileStream(
+                       cntlr,
+                       ugt_default_dimensions_json_file,
+                       'rt',
+                       encoding='utf-8'
+                )
+
+                val.usgaapDefaultDimensions = json.load(file)
+                file.close()
+
+            except FileNotFoundError:
+                if file:
+                    file.close()
+        year += 1
+
+
+def fire_dqc_us_0041_errors(val):
+    """
+    Fires all the dqc_us_0041 errors returned by _catch_dqc_us_0041_errors
+
+    :param val: ValidateXbrl to check if it contains errors
+    :type val: :class:'~arelle.ValidateXbrl.ValidateXbrl'
+    :return: No explicit return, but it fires all the dqc_us_0041 errors
+    :rtype: None
+    """
+    _setup_cache(val)
+
+    for error_info in _catch_dqc_us_0041_errors(val):
+        axis_name, axis_default_name, def_name = error_info
+        val.modelXbrl.error(
+            '{}.16'.format(_CODE_NAME),
+            messages.get_message(_CODE_NAME),
+            axis_name=axis_name,
+            axis_default_name=axis_default_name,
+            def_name=def_name,
+            ruleVersion=_RULE_VERSION
+        )
+
+
+def _default_dimension_mismatch(default_dimension, usgaap_default_dimensions):
+    """
+    Returns true if the default dimension is not included in the usgaap default
+    dimensions
+
+    :param default_dimension: dimension to test if it is a usgaap default
+        dimension
+    :type: str
+    :param usgaap_default_dimensions: list of usgaap taxonomies default
+        dimensions
+    :type: list
+    :return: True if the default dimensions is not included in the usgaap
+        default dimensions
+    :rtype: bool
+    """
+    if default_dimension != usgaap_default_dimensions:
+        return True
+    return False
+
+
+def _catch_dqc_us_0041_errors(val):
+    """
+    Returns a tuple containing the parts of the dqc_us_0041 error to be
+    displayed
+
+    :param val: ValidateXbrl to check if it contains errors
+    :type val: :class:'~arelle.ValidateXbrl.ValidateXbrl'
+    :return: all dqc_us_0041 errors
+    :rtype: tuple
+    """
+    _load_cache(val)
+    rel_set = val.modelXbrl.relationshipSet(
+        XbrlConst.dimensionDefault
+    ).modelRelationships
+    for rel in rel_set:
+        rel_to = rel.toModelObject
+        rel_from = rel.fromModelObject
+        if (_default_dimension_mismatch(
+                rel_to.name,
+                val.usgaapDefaultDimensions[rel_from.name]
+        )):
```

Comment:
SG: Multiline conditional formatting ([ref](https://github.com/Workiva/styleguide/blob/master/PYTHON.rst#multi-line-ifelifwith-statements)), suggestion:
```
if _default_dimension_mismatch(
    rel_to.name,
    val.usgaapDefaultDimensions[rel_from.name]
):
```
This avoids the ambiguous indent problem, hence the only one set of `( )`